### PR TITLE
Added types for chrome-har-capturer

### PR DESCRIPTION
**Pull request by NicolasNewman**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/chrome-har-capturer/package.json

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b446f5e860fe05d40aca8451d8baf3966676ea31
</p>

File: types/chrome-har-capturer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+export interface LoaderEventCallback<U> {
+    "load": (url: string, index: number, urls: readonly string[]) => void;
+    "done": (url: string, index: number, urls: readonly string[]) => void;
+    "fail": (url: string, err: any, index: number, urls: readonly string[]) => void;
+    "har": (har: HAR<U>) => void;
+}
+
+export interface RunOptions<T> {
+    /** Chrome Debugging Protocol host. Defaults to localhost */
+    host?: string;
```
Comment (jakebailey) [inline types/chrome-har-capturer/index.d.ts:107]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

File: types/chrome-har-capturer/index.d.ts [hunk 2]
```diff
@@ -136,7 +136,7 @@ export interface RunOptions<T> {
      * @param urls input URL array
      * @returns Promise executed before each page load
      */
-    preHook?: (url: string, client: any, index: number, urls: readonly string[]) => Promise<void>;
+    preHook?: (url: string, client: any, index: number, urls: readonly string[]) => Promise<void> | undefined;;
```
Comment (jakebailey) [inline types/chrome-har-capturer/index.d.ts:139]: I think you meant:

```suggestion
    preHook?: ((url: string, client: any, index: number, urls: readonly string[]) => Promise<void>) | undefined;
```

File: types/chrome-har-capturer/index.d.ts [hunk 3]
```diff
@@ -147,7 +147,7 @@ export interface RunOptions<T> {
      * @param urls input URL array
      * @returns Promise executed after each page load event
      */
-    postHook?: (url: string, client: any, index: number, urls: readonly string[]) => Promise<T>;
+    postHook?: (url: string, client: any, index: number, urls: readonly string[]) => Promise<T> | undefined;;
```
Comment (jakebailey) [inline types/chrome-har-capturer/index.d.ts:150]:
```suggestion
    postHook?: ((url: string, client: any, index: number, urls: readonly string[]) => Promise<T>) | undefined;
```

File: types/chrome-har-capturer/index.d.ts (hunk 2 above)
Comment (NicolasNewman) [reply types/chrome-har-capturer/index.d.ts]: Yes, got duplicated while pasting within GitHub

File: types/chrome-har-capturer/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    /**
+     * function returning a Promise executed before each page load
+     *
+     * @param url the current URL
+     * @param client CDP client instance
+     * @param index index of url in urls
+     * @param urls input URL array
+     * @returns Promise executed before each page load
+     */
+    preHook?: (url: string, client: any, index: number, urls: readonly string[]) => Promise<void> | undefined;
```
Comment (jakebailey) [inline types/chrome-har-capturer/index.d.ts:139]:
```suggestion
    preHook?: ((url: string, client: any, index: number, urls: readonly string[]) => Promise<void>) | undefined;
```

File: types/chrome-har-capturer/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
+/// <reference types="node" />
+
+import { EventEmitter } from "events";
+
+export interface Page<U> {
+    id: string;
+    title: any;
```
Comment (jakebailey) [inline types/chrome-har-capturer/index.d.ts:7]: It does seem like all of these `anys` could have proper types, but it isn't a complete requirement.

**Review by jakebailey — APPROVED** [review]

**Comment by NicolasNewman** [comment]
Ready to merge

**Pull request merged**